### PR TITLE
Fix a pthread_t handle leak #114610

### PR DESCRIPTION
**Pull request by g0djan**
https://github.com/rust-lang/rust/issues/114610

Ran the tests as described in https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/platform-support/wasm32-wasi-preview1-threads.md?plain=1#L125

### CODE REVIEW

File: library/std/src/sys/wasi/thread.rs [hunk 1]
```diff
@@ -178,6 +179,17 @@ impl Thread {
     }
 }
 
+cfg_if::cfg_if! {
```
Comment (Mark-Simulacrum) [inline library/std/src/sys/wasi/thread.rs:182]: Is there a reason not to put this alongside the definition of Thread, i.e., in that same cfg_if?

Comment (g0djan) [reply library/std/src/sys/wasi/thread.rs]: Moved it to the same cfg_if

**Review by programmerjake — APPROVED** [review]

**Comment by g0djan** [comment]
@Mark-Simulacrum is there anything else to address?

**Comment by wesleywiser** [comment]
Nominating for beta backport so that this fix can ride the release train with the `wasm32-wasi-preview1-threads` target which is newly Tier 2 in 1.73.

**Comment by cuviper** [comment]
Beta-backport approved today on [zulip](https://rust-lang.zulipchat.com/#narrow/stream/259402-t-libs.2Fmeetings/topic/Meeting.202023-09-20/near/392135477).

@rustbot label +beta-accepted

**Pull request merged**